staging/deploy.sh

```diff
@@ -6,11 +6,21 @@ set -eo pipefail
 DIR=$(git rev-parse --show-toplevel)
 VERB=${1:-switch}
 
-# Perform a dry-activation first.
-echo "dry-activating the configuration first..."
-nixos-rebuild dry-activate -I nixos-config=$DIR/staging/configuration.nix --target-host root@sectracker.nixpkgs.lahfa.xyz
+if [[ "$VERB" != "build" ]]; then
+  # Perform a dry-activation first.
+  echo "dry-activating the configuration first..."
+  nixos-rebuild dry-activate -I nixos-config=$DIR/staging/configuration.nix --target-host root@sectracker.nixpkgs.lahfa.xyz
+else
+  echo "skipping the dry-activation as we are using an offline verb."
+fi
 
-# This requires IPv6 to work as SSH is only IPv6-only.
-# Sorry, not sorry.
-echo "$VERB-ing the configuration now."
-nixos-rebuild $VERB -I nixos-config=$DIR/staging/configuration.nix --target-host root@sectracker.nixpkgs.lahfa.xyz
+
+if [[ "$VERB" != "build" ]]; then
```

Comment:
We're checking this twice now. Typo or why doesn't the work happen in the same conditional?

Reply:
Well these are two different phases, the reason for the second conditional is actually a bug in `nixos-rebuild` which performs a non-necessary copy-closure to the target host in presence of that flag.

We could log it as a bug but I really don't want to interact with the nonexistent upstream of these tools as you may know :).

Reply:
I mean why don't we

```bash
if $VERB == build then
  echo "skipping dry-activate for local build"
  nixos-rebuild build ...
else
  nixos-rebuild dry-activate ...
  nixos-rebuild $VERB --target-host ...
fi
```

Reply:
because theoretically we only need to exclude dry-activate, but this is a bug in presence of --target-host
I can implement what you just wrote, but this is just semantically incorrect

Reply:
Please do that or explain in a comment why the code looks differently from what one would expect; whatever you find easier :)

Reply:
ack